tests/accuracy_utils.py

```diff
@@ -80,26 +75,30 @@ def to_reference(inp, upcast=False):
     if inp is None:
         return None
     ref_inp = inp
-    if TO_CPU:
-        ref_inp = ref_inp.to("cpu")
     if upcast:
         ref_inp = ref_inp.to(torch.float64)
+    if TO_CPU:
+        ref_inp = ref_inp.to("cpu")
     return ref_inp
 
 
-def gems_assert_close(a, b, dtype, equal_nan=False, reduce_dim=1):
+def to_cpu(res, ref):
     if TO_CPU:
-        a = a.to("cpu")
-    b = b.to(dtype)
-    atol = 1e-4 * reduce_dim
-    rtol = RESOLUTION[dtype]
-    torch.testing.assert_close(a, b, atol=atol, rtol=rtol, equal_nan=equal_nan)
+        res = res.to("cpu")
+        assert ref.device == torch.device("cpu")
+    return res
 
 
-def gems_assert_equal(a, b):
-    if TO_CPU:
-        a = a.to("cpu")
-    assert torch.equal(a, b)
+def gems_assert_close(res, ref, dtype, equal_nan=False, reduce_dim=1):
```

Comment:
what's different between this and testing.assert_close?

Reply:
> what's different between this and testing.assert_close?

TO_CPU here.